### PR TITLE
Add doc for `mapInArrow` [skip ci]

### PR DESCRIPTION
**Pull request by firestarman**
Add doc for the Pyspark API `mapInArrow`, along with some doc updates for the Pandas UDF.

closes https://github.com/NVIDIA/spark-rapids/issues/6975

Signed-off-by: Firestarman <firestarmanllc@gmail.com>

### CODE REVIEW

**Comment by firestarman** [comment]
build

File: docs/additional-functionality/rapids-udfs.md [hunk 1]
```diff
@@ -286,3 +283,8 @@ The following configuration settings are also relevant for GPU scheduling for Pa
 To find details on the above Python configuration settings, please see the 
 [RAPIDS Accelerator for Apache Spark Configuration Guide](../configs.md). 
 Search 'pandas' for a quick navigation jump.
+
+
+### mapInArrow
+[mapInArrow](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.DataFrame.mapInArrow.html#pyspark.sql.DataFrame.mapInArrow) is a PySpark API introduced in Spark 3.3.0.
+And the RAPIDS Accelerator also supports acceleration for it in the same way with the Pandas UDF.
```
Comment (sameerz) [inline docs/additional-functionality/rapids-udfs.md:290]:
```suggestion
The RAPIDS Accelerator supports acceleration for `mapInArrow` in the same way with the Pandas UDF.
```

**Review by sameerz — COMMENTED** [review]
Minor nit

**Comment by firestarman** [comment]
build

File: docs/additional-functionality/rapids-udfs.md [hunk 2]
```diff
@@ -286,3 +283,8 @@ The following configuration settings are also relevant for GPU scheduling for Pa
 To find details on the above Python configuration settings, please see the 
 [RAPIDS Accelerator for Apache Spark Configuration Guide](../configs.md). 
 Search 'pandas' for a quick navigation jump.
+
+
+### mapInArrow
+[mapInArrow](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.DataFrame.mapInArrow.html#pyspark.sql.DataFrame.mapInArrow) is a PySpark API introduced in Spark 3.3.0.
+The RAPIDS Accelerator supports acceleration for `mapInArrow` in the same way with the Pandas UDF.
```
Comment (sameerz) [inline docs/additional-functionality/rapids-udfs.md:290]: Sorry, one more minor suggestion: 
```suggestion
The RAPIDS Accelerator supports acceleration for `mapInArrow` in the same way as Pandas UDF.
```

**Comment by firestarman** [comment]
build

**Review by sameerz — APPROVED** [review]

**Pull request merged**